src/display/graphics.c

```diff
@@ -0,0 +1,387 @@
+	static V5_TouchStatus v5_touch_status;
+	mutex_take(_display_mutex, TIMEOUT_MAX);
+	vexTouchDataGet(&v5_touch_status);
+	mutex_give(_display_mutex);
+	return v5_touch_status.lastEvent;
+}
+
+static linked_list_s_t* _touch_event_release_handler_list = NULL;
+static linked_list_s_t* _touch_event_press_handler_list = NULL;
+static linked_list_s_t* _touch_event_press_auto_handler_list = NULL;
+
+void _set_up_touch_callback_storage() {
+	_touch_event_release_handler_list = linked_list_init();
+	_touch_event_press_handler_list = linked_list_init();
+	_touch_event_press_auto_handler_list = linked_list_init();
+}
+
+void register_touch_callback(touch_event_cb_fn_t cb, touch_event_e_t event_type) {
+	switch (event_type) {
+	case E_TOUCH_EVENT_RELEASE:
+		linked_list_prepend_func(_touch_event_release_handler_list, (generic_fn_t)cb);
+		break;
+	case E_TOUCH_EVENT_PRESS:
+		linked_list_prepend_func(_touch_event_press_handler_list, (generic_fn_t)cb);
+		break;
+	case E_TOUCH_EVENT_PRESS_AND_HOLD:
+		linked_list_prepend_func(_touch_event_press_auto_handler_list, (generic_fn_t)cb);
+		break;
+	}
+}
+
+static task_stack_t touch_handle_task_stack[TASK_STACK_DEPTH_DEFAULT];
+static static_task_s_t touch_handle_task_buffer;
+static task_t touch_handle_task;
+
+void _handle_cb(ll_node_s_t* current, void* data) {
+	((touch_event_cb_fn_t)(current->payload.func))(((touch_event_position_data_s_t*)(data))->x,
+	                                               ((touch_event_position_data_s_t*)(data))->y);
+}
+static inline bool _touch_status_equivalent(V5_TouchStatus x, V5_TouchStatus y) {
+	return (x.lastEvent == y.lastEvent) && (x.lastXpos == y.lastXpos) && (x.lastYpos == y.lastYpos);
+}
+
+void _touch_handle_task(void* ignore) {
+	V5_TouchStatus last, current;
+	while (true) {
+		vexTouchDataGet(&current);
+		if (!_touch_status_equivalent(current, last)) {
+			touch_event_position_data_s_t event_data = { .x = current.lastXpos, .y = current.lastYpos };
+			switch (current.lastEvent) {
+			case E_TOUCH_EVENT_RELEASE:
+				linked_list_foreach(_touch_event_release_handler_list, _handle_cb, (void*)&event_data);
+				break;
+			case E_TOUCH_EVENT_PRESS:
+				linked_list_foreach(_touch_event_press_handler_list, _handle_cb, (void*)&event_data);
+				break;
+			case E_TOUCH_EVENT_PRESS_AND_HOLD:
+				linked_list_foreach(_touch_event_press_auto_handler_list, _handle_cb,
+				                    (void*)&event_data);
+				break;
+			}
+			last = current;
+		}
+		delay(5);
+	}
+}
+
+void graphical_context_daemon_initialize(void) {
+	_display_mutex = mutex_create();
+	_set_up_touch_callback_storage();
+	touch_handle_task =
+	    task_create_static(_touch_handle_task, NULL, TASK_PRIORITY_MAX - 3, TASK_STACK_DEPTH_DEFAULT,
+	                       "PROS Graphics Touch Handler", touch_handle_task_stack, &touch_handle_task_buffer);
+}
+
+
+//Extra text Printing Functions (Uncomment if ever needed to be added into API)
+/*
+void display_printf(const int16_t line, const char* fmt, ...) {
```

Comment:
If you don't have the symbols behind them defined, don't put declarations of functions in the header even if they're commented out.